### ISSUE TITLE
Remoting Exception (3.7.0)

### ISSUE THREAD

**Issue by WayneHiller** [nunit/nunit-console#340]
I am running tests from a single test assembly locally. Windows 2012 x64. Some of the tests will run and then I randomly get this error. The error does not always occur on the same tests. I had this all running fine before on this same machine. Something changed somewhere. Any ideas as to where to start looking?

1) Error : 
An existing connection was forcibly closed by the remote host

Server stack trace: 
   at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size, SocketFlags socketFlags)
   at System.Runtime.Remoting.Channels.SocketStream.Read(Byte[] buffer, Int32 offset, Int32 size)
   at System.Runtime.Remoting.Channels.SocketHandler.ReadFromSocket(Byte[] buffer, Int32 offset, Int32 count)
   at System.Runtime.Remoting.Channels.SocketHandler.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.Runtime.Remoting.Channels.SocketHandler.ReadAndMatchFourBytes(Byte[] buffer)
   at System.Runtime.Remoting.Channels.Tcp.TcpSocketHandler.ReadAndMatchPreamble()
   at System.Runtime.Remoting.Channels.Tcp.TcpSocketHandler.ReadVersionAndOperation(UInt16& operation)
   at System.Runtime.Remoting.Channels.Tcp.TcpClientSocketHandler.ReadHeaders()
   at System.Runtime.Remoting.Channels.Tcp.TcpClientTransportSink.ProcessMessage(IMessage msg, ITransportHeaders requestHeaders, Stream requestStream, ITransportHeaders& responseHeaders, Stream& responseStream)
   at System.Runtime.Remoting.Channels.BinaryClientFormatterSink.SyncProcessMessage(IMessage msg)

Exception rethrown at [0]: 
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage reqMsg, IMessage retMsg)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgData, Int32 type)
   at NUnit.Engine.ITestEngineRunner.Run(ITestEventListener listener, TestFilter filter)
   at NUnit.Engine.Runners.ProcessRunner.RunTests(ITestEventListener listener, TestFilter filter)

At the end of the log it states.

No connection could be made because the target machine actively refused it 127.0.0.1:56347
Failed to stop the remote agent. No connection could be made because the target machine actively refused it 127.0.0.1:56347

**Comment by CharliePoole** [comment]
NUnit console is reporting the only error it knows about: that the agent process crashed. At the end of the log, it tries to tell the agent to stop, which is a bit silly since it already crashed.

The agent leaves behind no info about why it crashed by default.

IF it's possible to run your tests in the same process as the console runner, use the `--inprocess` in hopes that you'll see the actual exception that crashed the agent.

If that isn't possible or if it gives you no better information, use the `--trace:Verbose` option to find out what caused the error.

**Comment by WayneHiller** [comment]
Ok thanks I will try those switches. The crash only happens on my testing machine. I am running the same tests on my dev machine now and everything is fine. I am use cake.net to automate the test running and I think it will let me pass command line switches through.

**Comment by fizxmike** [comment]
I'm getting the same error. My hunch is it might be related to our corporate proxy which is probably blocking the ports nunit3-console is using? But proxy should be bypassed for localhost, unfortunately my network settings are locked down on my dev machine... any suggestions?

```
NUnit Console Runner 3.7.0
Copyright (c) 2017 Charlie Poole, Rob Prouse

Runtime Environment
   OS Version: Microsoft Windows NT 6.1.7601 Service Pack 1
  CLR Version: 4.0.30319.42000

Test Files
    OneSignNVTest.dll


Errors, Failures and Warnings

1) Error :
An existing connection was forcibly closed by the remote host

Server stack trace:
   at System.Net.Sockets.Socket.Receive(Byte[] buffer, Int32 offset, Int32 size,
 SocketFlags socketFlags)
   at System.Runtime.Remoting.Channels.SocketStream.Read(Byte[] buffer, Int32 of
fset, Int32 size)
   at System.Runtime.Remoting.Channels.SocketHandler.ReadFromSocket(Byte[] buffe
r, Int32 offset, Int32 count)
   at System.Runtime.Remoting.Channels.SocketHandler.Read(Byte[] buffer, Int32 o
ffset, Int32 count)
   at System.Runtime.Remoting.Channels.SocketHandler.ReadAndMatchFourBytes(Byte[
] buffer)
   at System.Runtime.Remoting.Channels.Tcp.TcpSocketHandler.ReadAndMatchPreamble
()
   at System.Runtime.Remoting.Channels.Tcp.TcpSocketHandler.ReadVersionAndOperat
ion(UInt16& operation)
   at System.Runtime.Remoting.Channels.Tcp.TcpClientSocketHandler.ReadHeaders()
   at System.Runtime.Remoting.Channels.Tcp.TcpClientTransportSink.ProcessMessage
(IMessage msg, ITransportHeaders requestHeaders, Stream requestStream, ITranspor
tHeaders& responseHeaders, Stream& responseStream)
   at System.Runtime.Remoting.Channels.BinaryClientFormatterSink.SyncProcessMess
age(IMessage msg)

Exception rethrown at [0]:
   at System.Runtime.Remoting.Proxies.RealProxy.HandleReturnMessage(IMessage req
Msg, IMessage retMsg)
   at System.Runtime.Remoting.Proxies.RealProxy.PrivateInvoke(MessageData& msgDa
ta, Int32 type)
   at NUnit.Engine.ITestEngineRunner.Run(ITestEventListener listener, TestFilter
 filter)
   at NUnit.Engine.Runners.ProcessRunner.RunTests(ITestEventListener listener, T
estFilter filter)

Test Run Summary
  Overall result: Failed
  Test Count: 0, Passed: 0, Failed: 0, Warnings: 0, Inconclusive: 0, Skipped: 0
  Start time: 2018-01-17 00:22:13Z
    End time: 2018-01-17 00:22:14Z
    Duration: 1.107 seconds

Results (nunit3) saved as TestResult.xml

No connection could be made because the target machine actively refused it 127.0
.0.1:49364
Failed to stop the remote agent. No connection could be made because the target
machine actively refused it 127.0.0.1:49364
```

**Comment by ChrisMaddock** [comment]
@WayneHiller  This might be fixed by @BlythMeister's temporary fix in #370.

Would you try the master build, version 3.9.0-dev-03932 or later. You can find this on out [MyGet Feed](https://www.myget.org/feed/nunit/package/nuget/NUnit.ConsoleRunner).

Let us know if you have a different problem.

**Comment by ChrisMaddock** [comment]
Closing this, as there's not enough information to understand/reproduce the problem. 

Happy to reopen if anyone has more to add. 🙂
